plugin/ochttp/client_test.go

```diff
@@ -119,3 +120,111 @@ func TestClient(t *testing.T) {
 		}
 	}
 }
+
+func BenchmarkTransportNoInstrumentation(b *testing.B) {
+	benchmarkClientServer(b, &ochttp.Transport{NoStats: true, NoTrace: true})
+}
+
+func BenchmarkTransportTraceOnly(b *testing.B) {
+	benchmarkClientServer(b, &ochttp.Transport{NoStats: true})
+}
+
+func BenchmarkTransportStatsOnly(b *testing.B) {
+	benchmarkClientServer(b, &ochttp.Transport{NoTrace: true})
+}
+
+func BenchmarkTransportAllInstrumentation(b *testing.B) {
+	benchmarkClientServer(b, &ochttp.Transport{})
+}
+
+func benchmarkClientServer(b *testing.B, transport http.RoundTripper) {
+	b.ReportAllocs()
+	ts := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		fmt.Fprintf(rw, "Hello world.\n")
+	}))
+	defer ts.Close()
+	var client http.Client
+	client.Transport = transport
+	b.ResetTimer()
+
+	for i := 0; i < b.N; i++ {
+		res, err := client.Get(ts.URL)
+		if err != nil {
+			b.Fatalf("Get: %v", err)
+		}
+		all, err := ioutil.ReadAll(res.Body)
+		res.Body.Close()
+		if err != nil {
+			b.Fatal("ReadAll:", err)
+		}
+		body := string(all)
+		if body != "Hello world.\n" {
+			b.Fatal("Got body:", body)
+		}
+	}
+}
+
+func BenchmarkTransportParallel64NoInstrumentation(b *testing.B) {
+	benchmarkClientServerParallel(b, 64, &ochttp.Transport{NoTrace: true, NoStats: true})
+}
+
+func BenchmarkTransportParallel64TraceOnly(b *testing.B) {
+	benchmarkClientServerParallel(b, 64, &ochttp.Transport{NoStats: true})
+}
+
+func BenchmarkTransportParallel64StatsOnly(b *testing.B) {
+	benchmarkClientServerParallel(b, 64, &ochttp.Transport{NoTrace: true})
+}
+
+func BenchmarkTransportParallel64AllInstrumentation(b *testing.B) {
+	benchmarkClientServerParallel(b, 64, &ochttp.Transport{})
+}
+
+func benchmarkClientServerParallel(b *testing.B, parallelism int, transport *ochttp.Transport) {
+	b.ReportAllocs()
+	ts := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		fmt.Fprintf(rw, "Hello world.\n")
+	}))
+	defer ts.Close()
+
+	var c http.Client
+	transport.Base = &http.Transport{
+		MaxIdleConns:        parallelism,
+		MaxIdleConnsPerHost: parallelism,
+	}
+	c.Transport = transport
+
+	b.ResetTimer()
+
+	// TODO(ramonza): replace with b.RunParallel (it didn't work when I tried)
+
+	var wg sync.WaitGroup
```

Comment:
Why didn't we just use a channel? I believe we can radically simplify the code below like this:

```go
semaphore := make(chan bool, concurrency)
for i := 0; i < b.N; i++ {
     <-semaphore
     go func() {
          defer func() { semaphore <- true }()
          res, err := c.Get(...)
     }()
}
```